### PR TITLE
odo release notes 4.4

### PR DESCRIPTION
**Pull request by yhontyk**
branch 4.4

### CODE REVIEW

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 1]
```diff
@@ -6,32 +6,48 @@ include::modules/common-attributes.adoc[]
 toc::[]
 
 [id="odo-notable-improvements_{context}"]
-== Notable improvements in `{odo-title}`{nbsp}{odo-ver}
+== Notable changes and improvements in `{odo-title}`{nbsp}{odo-ver}
+
+* A `--now` flag is added for `odo create`, `odo url` and `odo config`. 
++
+Run `odo url create --now` to create the URL in the configuration and propagate all the changes to the cluster
```
Comment (mburke5678) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:13]: Period at the end of _all the changes to the cluster_

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 2]
```diff
@@ -6,32 +6,48 @@ include::modules/common-attributes.adoc[]
...
+* An `odo debug info` command is added. The command displays the information whether the debug mode is enabled for the component, whether the port-forward process is running, and information about ports used.
+* `odo push` output now shows the exact error from the OpenShift cluster on failure.
```
Comment (mburke5678) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:15]: s/`odo push`/The `odo push`

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 3]
```diff
@@ -6,32 +6,48 @@ include::modules/common-attributes.adoc[]
 toc::[]
 
 [id="odo-notable-improvements_{context}"]
-== Notable improvements in `{odo-title}`{nbsp}{odo-ver}
+== Notable changes and improvements in `{odo-title}`{nbsp}{odo-ver}
+
+* A `--now` flag is added for `odo create`, `odo url` and `odo config`. 
++
+Run `odo url create --now` to create the URL in the configuration and propagate all the changes to the cluster
+* An `odo debug info` command is added. The command displays the information whether the debug mode is enabled for the component, whether the port-forward process is running, and information about ports used.
```
Comment (mburke5678) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:14]: The command displays shows if the debug mode is enabled for a component, if the port-forward process is running, and information about ports used.
Question: In 4.3, the ODO docs say that _Components created with odo run in the debug mode by default. _ Can the user turn off debug mode?

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 4]
```diff
@@ -6,32 +6,48 @@ include::modules/common-attributes.adoc[]
...
+* `odo push` output now shows the exact error from the OpenShift cluster on failure.
++
+* A `--secure` flag is enabled for `odo url`. It indicates whether a URL is secure. 
+* `odo storage list` now displays the information about the storage state: `Pushed`, `NotPushed` or `Locally Deleted`.
+* `odo debug port-forward` now selects a port automatically if the default port (5858) is occupied.
+* The `--all` flag is renamed to  `--all-apps`.
+* The default PVC size has been increased to 10GiB
```
Comment (mburke5678) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:21]: Period at the end.

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 5]
```diff
@@ -6,32 +6,48 @@ include::modules/common-attributes.adoc[]
...
+* The JSON output for `odo storage list -o json` has been restructured.
```
Comment (mburke5678) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:22]: Does the user need more information about the new structure?

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 6]
```diff
@@ -6,32 +6,48 @@ include::modules/common-attributes.adoc[]
 toc::[]
 
 [id="odo-notable-improvements_{context}"]
-== Notable improvements in `{odo-title}`{nbsp}{odo-ver}
+== Notable changes and improvements in `{odo-title}`{nbsp}{odo-ver}
+
+* A `--now` flag is added for `odo create`, `odo url` and `odo config`.
```
Comment (mburke5678) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:11]: comma after `odo url`

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 7]
```diff
@@ -6,32 +6,48 @@ include::modules/common-attributes.adoc[]
...
-== Notable improvements in `{odo-title}`{nbsp}{odo-ver}
+== Notable changes and improvements in `{odo-title}`{nbsp}{odo-ver}
+
+* A `--now` flag is added for `odo create`, `odo url` and `odo config`. 
++
+Run `odo url create --now` to create the URL in the configuration and propagate all the changes to the cluster
+* An `odo debug info` command is added. The command displays the information whether the debug mode is enabled for the component, whether the port-forward process is running, and information about ports used.
+* `odo push` output now shows the exact error from the OpenShift cluster on failure.
++
+* A `--secure` flag is enabled for `odo url`. It indicates whether a URL is secure.
```
Comment (mburke5678) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:17]: s/enabled/added ?
To match _A `--now` flag is added_
 https://github.com/openshift/openshift-docs/pull/20957/files#diff-c2001e8b2095b3f6a515b452c5c694f1R11

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 8]
```diff
@@ -6,32 +6,48 @@ include::modules/common-attributes.adoc[]
...
+== Notable changes and improvements in `{odo-title}`{nbsp}{odo-ver}
+
+* A `--now` flag is added for `odo create`, `odo url` and `odo config`. 
++
+Run `odo url create --now` to create the URL in the configuration and propagate all the changes to the cluster
+* An `odo debug info` command is added. The command displays the information whether the debug mode is enabled for the component, whether the port-forward process is running, and information about ports used.
+* `odo push` output now shows the exact error from the OpenShift cluster on failure.
++
+* A `--secure` flag is enabled for `odo url`. It indicates whether a URL is secure. 
+* `odo storage list` now displays the information about the storage state: `Pushed`, `NotPushed` or `Locally Deleted`.
```
Comment (mburke5678) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:18]: Will users know what these states mean? I did a quick search and didn't see these terms in the 4.3 docs.

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 9]
```diff
@@ -6,32 +6,48 @@ include::modules/common-attributes.adoc[]
...
+* `odo debug port-forward` now selects a port automatically if the default port (5858) is occupied.
+* The `--all` flag is renamed to  `--all-apps`.
+* The default PVC size has been increased to 10GiB
+* The JSON output for `odo storage list -o json` has been restructured.
+* `nodejs-8` and `nodejs-10` images are no longer supported.
+* Experimental mode feature. By default, features under development or in experimental mode are hidden from the user.
```
Comment (mburke5678) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:24]: Should this be in a New Features topic like in OCP release notes?
https://docs.openshift.com/container-platform/4.3/release_notes/ocp-4-3-release-notes.html#ocp-4-3-new-features-and-enhancements

**Comment by mburke5678** [comment]
@boczkowska A few nits and a couple of questions.

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc (hunk 3 above)
Comment (boczkowska) [reply cli_reference/openshift_developer_cli/odo-release-notes.adoc:14]: @mburke5678 yes, the users can do that. I will check the assembly about debugging and if it's not there and update it with the instruction how to do that.

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc (hunk 5 above)
Comment (boczkowska) [reply cli_reference/openshift_developer_cli/odo-release-notes.adoc:22]: Not really. It's mostly a cosmetic change to make it more readable.

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc (hunk 8 above)
Comment (boczkowska) [reply cli_reference/openshift_developer_cli/odo-release-notes.adoc:18]: These states are self-evident. More detailed explanation will go into storage documentation for odo which is still in progress.

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc (hunk 9 above)
Comment (boczkowska) [reply cli_reference/openshift_developer_cli/odo-release-notes.adoc:24]: I've sent a PR for odo paragraph for the general RNs here: https://github.com/openshift/openshift-docs/pull/21071

**Comment by boczkowska** [comment]
@mburke5678 Thank you for the review! I've addressed all of you feedback and answered your questions. Please take look when you have time. :blush:

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 10]
```diff
@@ -6,32 +6,48 @@ include::modules/common-attributes.adoc[]
 toc::[]
 
 [id="odo-notable-improvements_{context}"]
-== Notable improvements in `{odo-title}`{nbsp}{odo-ver}
+== Notable changes and improvements in `{odo-title}`{nbsp}{odo-ver}
+
+* A `--now` flag is added for `odo create`, `odo url`, and `odo config`. 
++
+Run `odo url create --now` to create the URL in the configuration and propagate all the changes to the cluster.
+* An `odo debug info` command is added. TThe command displays if the debug mode is enabled for a component, if the port-forward process is running, and information about ports used.
```
Comment (lamek) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:14]: Extra "T" on "TThe".

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 11]
```diff
@@ -42,36 +58,23 @@ If you find an error or have suggestions for improving the documentation, file a
 
 .For Product
 
-If you find an error, encountered a bug, or have suggestions for improving the functionality of `{odo-title}`, file an issue in link:http://bugzilla.redhat.com[Bugzilla]. Choose the *Red Hat odo for {product-title}* product type.
+If you find an error, encountered a bug, or have suggestions for improving the functionality of `{odo-title}`, file an issue in link:http://bugzilla.redhat.com[Bugzilla]. Choose the *Red Hat {odo-title} for {product-title}* product type.
```
Comment (lamek) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:61]: "encountered" -> "encounter" to keep consistency with tense.

File: cli_reference/openshift_developer_cli/odo-release-notes.adoc [hunk 12]
```diff
@@ -82,14 +85,25 @@ Provide as many details in the issue description as possible.
...
 :FeatureName: odo debug
 include::modules/technology-preview.adoc[leveloffset=+1]
 
-- Use the `odo debug port-forward` command to start port forwarding.
-- Use the `odo config set DebugPort 9292` command to specify the remote port that the debugging agent should run on.
-- Use the `odo debug port-forward --local-port 9292` command to specify the local port for port forwarding.
\ No newline at end of file
+* Devfile support. You can now create and deploy your applications using the devfiles with `{odo-title}`. A devfile is a file that defines the development environment: environment variables, image, and so on. To access this feature, you must enable experimental mode with `odo preference set experimental true`.
++
+To see the list currently supported devfile components, run `odo catalog list components`
```
Comment (lamek) [inline cli_reference/openshift_developer_cli/odo-release-notes.adoc:106]: Missing the word "of". "To see the list **of** currently supported devfile components"...

**Review by lamek — COMMENTED** [review]
Few small nits, otherwise LGTM.

**Comment by boczkowska** [comment]
@lamek thank you for your review! Fixed everything and squashed. :)

**Pull request merged**